snap/container_test.go

```diff
@@ -107,7 +107,7 @@ version: 1
 	c.Assert(err, IsNil)
 
 	err = snap.ValidateSnapContainer(s.container(), info, discard)
-	c.Check(err, Equals, snap.ErrBadModes)
+	c.Check(err, ErrorMatches, `snap is unusable due to bad permissions: "." should be world-readable and executable, and isn't: drwx------`)
```

Comment:
Perhaps add something that checks with the equivalent of errors.Is, for completeness.

Reply:
good point, thanks! I added those to all relevant tests